tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -140,13 +143,70 @@ void sem_take_multiple_high_prio_helper(void *p1, void *p2, void *p3)
 	k_sem_give(&high_prio_sem);
 }
 
+/* First function for mutual exclusion test */
+void sem_queue_mutual_exclusion1(void *p1, void *p2, void *p3)
+{
+	for (int i = 0; i < 5; i++) {
+		k_sem_take(&mut_sem, K_FOREVER);
+
+		/* in that function critical section makes critical var +1 */
+		critical_var += 1;
```

Comment:
Has this been initialized to 0? Maybe it is set implicitly with the definition, but perhaps it could be improved.

Reply:
You noticed weak point. Thank you.